packages/core/src/sync-store/postgres/store.ts

```diff
@@ -2202,114 +2215,35 @@ export class PostgresSyncStore implements SyncStore {
         });
       });
 
-      const hasNextPage = events.length === limit + 1;
-
-      if (!hasNextPage) {
-        yield events;
-        break;
+      // set fromCursor + seconds
+      if (events.length === 0) {
+        this.seconds = Math.round(this.seconds * 2);
```

Comment:
Could consider making these constants explicit options, but fine to skip for now I think.